astropy/units/tests/test_units.py

```diff
@@ -292,62 +292,78 @@ def test_empty_compose():
         composed = u.m.compose(units=[])
 
 
-def test_compose_roundtrip():
-    def _test_compose_roundtrip(unit):
-        composed_list = unit.decompose().compose()
-        found = False
-        for composed in composed_list:
-            if len(composed.bases):
-                if composed.bases[0] is unit:
-                    found = True
-                    break
-            elif len(unit.bases) == 0:
+def _unit_as_str(unit):
+    # This function serves two purposes - it is used to sort the units to
+    # test alphabetically, and it is also use to allow pytest to show the unit
+    # in the [] when running the parametrized tests.
+    return str(unit)
+    
+    
+# We use a set to make sure we don't have any duplicates.
+COMPOSE_ROUNDTRIP = set()
```

Comment:
Now that we are taking this out for real, is it cleaner to have this in `setup_module()` for `pytest`? However, when I tried that with my version, the list (or set here) only picked up *one* value instead of the entire list. So maybe it is not worth the trouble.